p2p/src/network/select/p2p_network_select_effects.rs

```diff
@@ -92,104 +92,117 @@ impl P2pNetworkSelectAction {
-                                                    addr,
-                                                    peer_id,
-                                                    stream_id,
-                                                },
-                                            );
-                                        }
-                                    }
-                                    StreamKind::Identify(IdentifyAlgorithm::Identify1_0_0) => {
-                                        if !fin {
-                                            //println!("==== {}", hex::encode(&a.data.0));
-                                            store.dispatch(
-                                                P2pNetworkIdentifyStreamAction::IncomingData {
-                                                    addr,
-                                                    peer_id,
-                                                    stream_id,
-                                                    data,
-                                                },
-                                            );
-                                        } else {
-                                            store.dispatch(
-                                                P2pNetworkIdentifyStreamAction::RemoteClose {
-                                                    addr,
-                                                    peer_id,
-                                                    stream_id,
-                                                },
-                                            );
-                                        }
-                                    }
-                                    StreamKind::Identify(IdentifyAlgorithm::IdentifyPush1_0_0) => {
-                                        //unimplemented!()
-                                    }
-                                    StreamKind::Broadcast(_) => {
-                                        store.dispatch(P2pNetworkPubsubAction::IncomingData {
-                                            peer_id,
-                                            addr,
-                                            stream_id,
-                                            data,
-                                            seen_limit: store.state().config.meshsub.mcache_len,
-                                        });
-                                    }
-                                    StreamKind::Ping(PingAlgorithm::Ping1_0_0) => {
-                                        //unimplemented!()
-                                    }
-                                    StreamKind::Bitswap(_) => {
-                                        //unimplemented!()
-                                    }
-                                    StreamKind::Status(_) => {
-                                        //unimplemented!()
+                let Some(Some(negotiated)) = &state.negotiated else {
```

Comment:
much better this way, now the error is immediately after the failed condition, and not after the happy-path block